client/App/Services/WifiService.js

```diff
@@ -18,8 +18,7 @@ async function fetchWifiList() {
 async function getWifiDataForSample(wifiList) {
   // TODO: if imei is here to stay, it should be retrieved
   // on startup once, and saved to store!
-  const imeis = await IMEI.getImei();
-  const imei = imeis.length ? imeis[0] : null;
+  const imei = device.getUniqueId(); 
```

Comment:
Did you check this on non-android 10 device? If so - awesome. If not - let me know and I'll check:)